packages/data-context/test/unit/sources/FileDataSource.spec.ts

```diff
@@ -0,0 +1,113 @@
+import * as fileUtil from '../../../src/util/file'
+
+describe('FileDataSource', () => {
+  let projectPath: string
+  let scriptsFolder: string
+  let ctx: DataContext
+  let fileDataSource: FileDataSource
+
+  beforeEach(async () => {
+    projectPath = await scaffoldProject('globby-test-bed')
+    scriptsFolder = path.join(projectPath, 'scripts')
+
+    ctx = createTestDataContext('open')
+    ctx.coreData.currentTestingType = 'e2e'
+
+    fileDataSource = new FileDataSource(ctx)
+  })
+
+  afterEach(() => {
+    removeProject('globby-test-bed')
+    sinon.restore()
+  })
+
+  describe('#getFilesByGlob', () => {
+    it('finds files at root matching given pattern using globby', async () => {
+      const files = await fileDataSource.getFilesByGlob(
+        projectPath,
+        'root-script-*.js',
+      )
+
+      expect(files).to.have.length(2)
+    })
+
+    it('finds files matching relative patterns in working dir', async () => {
+      const files = await fileDataSource.getFilesByGlob(
+        projectPath,
+        './root-script-*.js',
+      )
+
+      expect(files).to.have.length(2)
+    })
+
+    it('finds files matching patterns that include working dir', async () => {
+      const files = await fileDataSource.getFilesByGlob(
+        projectPath,
+        `${projectPath}/root-script-*.js`,
+      )
+
+      expect(files).to.have.length(2)
+    })
+
+    it('finds files matching multiple patterns', async () => {
+      const files = await fileDataSource.getFilesByGlob(
+        projectPath,
+        ['root-script-*.js', 'scripts/**/*.js'],
+      )
+
+      expect(files).to.have.length(5)
+    })
+
+    it('does not find files outside of working dir', async () => {
+      const files = await fileDataSource.getFilesByGlob(
+        scriptsFolder,
+        ['root-script-*.js', './**/*.js'],
+      )
+
+      expect(files).to.have.length(3)
+    })
+
+    it('always ignores files within node_modules', async () => {
+      const nodeModulesPath = path.join(projectPath, 'node_modules')
+
+      await fs.mkdir(nodeModulesPath)
+      await fs.writeFile(path.join(nodeModulesPath, 'module-script-1.js'), '')
+      await fs.writeFile(path.join(nodeModulesPath, 'module-script-2.js'), '')
+
+      const files = await fileDataSource.getFilesByGlob(
+        projectPath,
+        '**/*script-*.js',
+        { ignore: ['./scripts/**/*'] },
+      )
+
+      // only scripts at root should be found, as node_modules is implicitly ignored
+      // and ./scripts is explicitly ignored
+      expect(files).to.have.length(2)
+    })
+
+    it('converts globs to POSIX paths on windows', async () => {
+      const windowsSeperator = '\\'
+
+      sinon.stub(os, 'platform').returns('win32')
+      const toPosixStub = sinon.stub(fileUtil, 'toPosix').callsFake((path) => {
+        return toPosixStub.wrappedMethod(path, windowsSeperator)
+      })
+
+      const files = await fileDataSource.getFilesByGlob(
+        projectPath,
```

Comment:
we we actually glob by the project path? I often see this error:

<img width="893" alt="Screen Shot 2022-08-09 at 12 57 51 PM" src="https://user-images.githubusercontent.com/14099737/184253746-f04e01c7-41e1-43a2-9484-37f5be133b61.png">

Reply:
The [error itself](https://github.com/cypress-io/cypress/blob/2cfd4a95c804ad763d5cb037e83f780ad09c44be/packages/errors/src/errors.ts#L532-L557) has special formatting to show both the search directory and the glob pattern in those contrasting colors. Which was misleading in the case of these `--spec` options prior to this PR, as the glob pattern we were using really _was_ the entire absolute path.